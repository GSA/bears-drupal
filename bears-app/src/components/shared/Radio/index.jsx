import { Label } from '../index'
import PropTypes from 'prop-types'

const Radio = ({ label, value, defaultChecked }) => {
  return (
    <>
      <div className="usa-radio">
        <input
          className="usa-radio__input"
          id={label}
          type="radio"
          name={label}
          value={value || label}
          defaultChecked={defaultChecked || false}
        />
        <Label className="usa-radio__label" htmlFor={label} label={label} />
      </div>
    </>
  )
}

Radio.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  defaultChecked: PropTypes.bool,
}

export default Radio
