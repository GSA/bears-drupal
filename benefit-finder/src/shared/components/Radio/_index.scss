@use '../../styles/colors/_index.scss' as color;
@use '../../styles/space/_index.scss' as space;

#benefit-finder {
  .usa-radio__label {
    padding-bottom: space.$padding-bottom-md-plus;
  }

  .usa-input--error + .usa-radio__label::before {
    box-shadow: 0 0 0 3px color.$alert-red;
  }
}
