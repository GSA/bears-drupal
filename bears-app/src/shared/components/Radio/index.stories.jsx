import Radio from './index.jsx'

export default {
  component: Radio,
  tags: ['autodocs'],
  args: {
    label: 'Radio',
    value: 'radio',
    defaultChecked: false,
  },
}

export const Primary = {}

export const DefaultChecked = {
  args: {
    defaultChecked: true,
  },
}
