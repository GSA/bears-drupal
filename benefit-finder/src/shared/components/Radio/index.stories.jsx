import Radio from './index.jsx'

export default {
  component: Radio,
  tags: ['autodocs'],
  args: {
    label: 'Radio',
    value: 'radio',
    checked: false,
  },
}

export const Primary = {}

export const DefaultChecked = {
  args: {
    checked: true,
  },
}
