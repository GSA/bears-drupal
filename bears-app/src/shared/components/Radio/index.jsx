import { Label } from '../index'
import PropTypes from 'prop-types'

import './_index.scss'

/**
 * a functional component that renders an input of radio type
 * @component
 * @param {string} label - displayed in ui
 * @param {string} value - assigned to value param in html
 * @param {boolean} defaultChecked - determines if the radio is default selected
 * @return {html} returns a semantic input as type radio element
 */

const Radio = ({ label, value, checked, onChange, requiredLabel }) => {
  // console.log(handleErrorState)
  return (
    <>
      <div className="usa-radio">
        <input
          className="usa-radio__input"
          id={label}
          type="radio"
          name={label}
          value={value || label}
          checked={checked}
          onChange={onChange}
        />
        <Label className="usa-radio__label" htmlFor={label} label={label} />
      </div>
    </>
  )
}

Radio.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  defaultChecked: PropTypes.bool,
}

export default Radio
