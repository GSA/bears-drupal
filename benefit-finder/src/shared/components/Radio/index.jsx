import { useHandleClassName } from '../../hooks/useHandleClassName'
import { Label } from '../index'
import PropTypes from 'prop-types'

import './_index.scss'

/**
 * a functional component that renders an input of radio type
 * @component
 * @param {string} label - displayed in ui
 * @param {string} value - assigned to value param in html
 * @param {boolean} checked - determines if the radio is selected
 * @param {func} onChange - inherited onChange handler
 * @param {string} classNAme - inherited class strings
 * @return {html} returns a semantic input as type radio element
 */

const Radio = ({
  id,
  label,
  value,
  checked,
  onChange,
  required,
  className,
  name,
}) => {
  const handleRequired = required === 'TRUE' ? ['required-field'] : ''
  const defaultClasses = ['usa-radio__input']
  const utilityClasses = handleRequired
  return (
    <>
      <div className="usa-radio">
        <input
          className={useHandleClassName({
            className,
            defaultClasses,
            utilityClasses,
          })}
          type="radio"
          name={name}
          value={value || id}
          checked={checked}
          onChange={onChange}
        />
        <Label className="usa-radio__label" htmlFor={id} label={label} />
      </div>
    </>
  )
}

Radio.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  defaultChecked: PropTypes.bool,
}

export default Radio
