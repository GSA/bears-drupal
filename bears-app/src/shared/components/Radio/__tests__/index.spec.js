import renderer from 'react-test-renderer'
import Radio from '../index.jsx'

describe('Radio', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<Radio />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
