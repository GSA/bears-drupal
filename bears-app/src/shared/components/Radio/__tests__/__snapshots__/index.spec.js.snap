// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders a match to the previous snapshot 1`] = `
<div
  className="usa-radio"
>
  <input
    className=""
    type="radio"
  />
  <label
    className=""
  />
</div>
`;
